cognite/neat/rules/models/_rules/dms_schema.py

```diff
@@ -67,6 +85,171 @@ def from_model_id(cls, client: CogniteClient, data_model_id: dm.DataModelIdentif
+        data_models.mkdir(parents=True, exist_ok=True)
+        if "spaces" not in exclude_set:
+            for space in self.spaces:
+                (data_models / f"{space.space}.space.yaml").write_text(space.dump_yaml())
+        if "data_models" not in exclude_set:
+            for model in self.data_models:
+                (data_models / f"{model.external_id}.datamodel.yaml").write_text(model.dump_yaml())
+        if "views" not in exclude_set and self.views:
+            view_dir = data_models / "views"
+            view_dir.mkdir(parents=True, exist_ok=True)
+            for view in self.views:
+                (view_dir / f"{view.external_id}.view.yaml").write_text(view.dump_yaml())
+        if "containers" not in exclude_set and self.containers:
+            container_dir = data_models / "containers"
+            container_dir.mkdir(parents=True, exist_ok=True)
+            for container in self.containers:
+                (container_dir / f"{container.external_id}.container.yaml").write_text(container.dump_yaml())
+        if "node_types" not in exclude_set and self.node_types:
+            node_dir = data_models / "nodes"
+            node_dir.mkdir(parents=True, exist_ok=True)
+            for node in self.node_types:
+                (node_dir / f"{node.external_id}.node.yaml").write_text(node.dump_yaml())
+
+    @classmethod
+    def from_zip(cls, zip_file: str | Path) -> Self:
+        """Load a schema from a ZIP file containing YAML files.
+
+        The ZIP file is expected to follow the Cognite-Toolkit convention
+        where each file is named as `resource_type.resource_name.yaml`.
+        """
+        data = cls._read_zip(Path(zip_file))
+        return cls.load(data)
+
+    @classmethod
+    def _read_zip(cls, zip_file: Path) -> dict[str, list[Any]]:
+        data: dict[str, list[Any]] = {}
+        with zipfile.ZipFile(zip_file, "r") as zip_ref:
+            for file_info in zip_ref.infolist():
+                if file_info.filename.endswith(".yaml"):
+                    if "/" not in file_info.filename:
+                        continue
+                    filename = Path(file_info.filename.split("/")[-1])
+                    if "." not in filename.stem:
+                        continue
+                    resource_type = filename.stem.rsplit(".", 1)[-1]
+                    if attr_name := cls._FIELD_NAME_BY_RESOURCE_TYPE.get(resource_type):
+                        data.setdefault(attr_name, [])
+                        loaded = yaml.safe_load(zip_ref.read(file_info).decode())
+                        if isinstance(loaded, list):
+                            data[attr_name].extend(loaded)
+                        else:
+                            data[attr_name].append(loaded)
+        return data
+
+    def to_zip(self, zip_file: str | Path, exclude: set[str] | None = None) -> None:
+        """Save the schema to a ZIP file as YAML files. This is compatible with the Cognite-Toolkit convention.
```

Comment:
is this toolkit format? if yes, add note

Reply:
`This is compatible with the Cognite-Toolkit convention.`

Yes, and note is there.